server/src/main/java/com/project/cisco/service/impl/MessageServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+package com.project.cisco.service.impl;
+
+import com.project.cisco.database.entity.Language;
+import com.project.cisco.database.entity.Message;
+import com.project.cisco.database.entity.Tag;
+import com.project.cisco.database.repository.LanguageRepository;
+import com.project.cisco.database.repository.MessageRepository;
+import com.project.cisco.database.repository.TagRepository;
+import com.project.cisco.dto.MessageDto;
+import com.project.cisco.exception.NotAllowedLanguageException;
+import com.project.cisco.exception.NotFoundException;
+import com.project.cisco.exception.UniqueConstraintViolationException;
+import com.project.cisco.mapper.MessageMapper;
+import com.project.cisco.service.MessageService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+public class MessageServiceImpl implements MessageService {
+
+    @Autowired
+    private LanguageRepository languageRepository;
+    @Autowired
+    private TagRepository tagRepository;
+    @Autowired
+    private MessageRepository messageRepository;
+    @Autowired
+    private MessageMapper messageMapper;
+
+    @Override
+    public MessageDto addMessage(MessageDto messageDto) {
```

Comment:
We have a use case specifying that: "Translation should have same tags as original message". I believe that should be on a backend side, to check if `original_message` has any tags and copy them to the new one.